### PR TITLE
Don't auto-disable OpenGL for debugger-enabled builds on Linux

### PR DESCRIPTION
**Pull request by kcgen**
Now that @GranMinigun has fixed OpenGL when used with the debugger, this PR reverts the exclusion of OpenGL in debugger builds.

### CODE REVIEW

**Comment by kcgen** [comment]
Allowing OpenGL in debugger-enabled builds opened up more code paths (which in turn resulted in more warnings), so this PR had to squash those.

@GranMinigun , I tried to be as minimal as possible to resude the rebase-pain for both your active sdlmain PRs.

File: meson.build [hunk 1]
```diff
@@ -549,29 +549,9 @@ summary('SDL_image 2.0 support', sdl2_image_dep.found())
...
-
-    warning(
-        '''OpenGL is disabled because it's not compatible with the debugger.
-            To use OpenGL with the debugger, force it with "-Duse_opengl=true"''',
-    )
-
-    conf_data.set10('C_OPENGL', false)
-    opengl_summary_msg = 'Found, but disabled due to conflict with debugger'
-else
-    conf_data.set10('C_OPENGL', opengl_dep.found())
```
Comment (GranMinigun) [inline meson.build:571]: This cuts out OpenGL completely.

Comment (kcgen) [reply meson.build:571]: In that case, we can get rid of a lot more code :laughing: j/k

(thanks for this big catch.. re-adding)

**Review by GranMinigun — APPROVED** [review]

**Comment by kcgen** [comment]
@GranMinigun , thanks for the checks.  Will merge this after the integer PR goes in.

**Pull request merged**